core/Stream/hsStream.cpp

```diff
@@ -381,83 +431,134 @@ bool hsFileStream::open(const ST::string& file, FileMode mode)
     } else if (mode == fmRead || mode == fmReadWrite) {
         throw hsFileReadException(__FILE__, __LINE__, "File does not exist");
     }
+#endif
     return false;
 }
 
 void hsFileStream::close()
 {
-    if (F)
+    if (F) {
+#ifdef _WIN32
+        CloseHandle(F);
+#else
         fclose(F);
+#endif
+    }
     F = nullptr;
 }
 
 uint32_t hsFileStream::size() const
 {
     if (F == nullptr)
         return 0;
+
+#ifdef _WIN32
+    auto sz = static_cast<uint32_t>(GetFileSize(F, nullptr));
+#else
     unsigned int p = ftell(F);
     fseek(F, 0, SEEK_END);
     unsigned int sz = ftell(F);
     fseek(F, p, SEEK_SET);
+#endif
     return sz;
 }
 
 uint32_t hsFileStream::pos() const
 {
     if (F == nullptr)
         return 0;
+#ifdef _WIN32
+    return SetFilePointer(F, 0, nullptr, FILE_CURRENT);
+#else
     return ftell(F);
+#endif
 }
 
 bool hsFileStream::eof() const
 {
     if (F == nullptr)
         return true;
+
+#ifdef _WIN32
+    uint8_t buf;
+    DWORD nread = 0;
+    BOOL result = ReadFile(F, &buf, 1, &nread, nullptr);
+    if (result == TRUE && nread == 0)
+        return true;
+    SetFilePointer(F, -1, nullptr, FILE_CURRENT);
```

Comment:
Hmm, this still probably won't work on non-seekable streams...  But I guess the Win32 API doesn't have any way to push a byte back into the read buffer like C stdio does :/.

Oh well, probably good enough for our purposes.

Reply:
Yeah, the only thing I saw for peeking data from a `HANDLE` was specifically for pipes 😞